services/terrain/src/terrain/clients/ezid.clj

```diff
@@ -0,0 +1,84 @@
+(ns ^{:author psarando} terrain.clients.ezid
+  (:use [ring.util.http-response :only [charset]]
+        [slingshot.slingshot :only [try+ throw+]])
+  (:require [cemerick.url :as curl]
+            [clj-http.client :as http]
+            [clojure.string :as string]
+            [terrain.util.config :as config]))
+
+(defn- auth-params
+  []
+  [(config/ezid-username) (config/ezid-password)])
+
+(defn- ezid-uri
+  [& components]
+  (str (apply curl/url (config/ezid-base-url) components)))
+
+(defn- anvl-escape
+  [value]
+  (-> value
+      (string/replace #"%"  "%25")
+      (string/replace #"\n" "%0A")
+      (string/replace #"\r" "%0D")
+      (string/replace #":"  "%3A")))
+
+(defn- anvl-unescape
+  [value]
+  (-> value
+      (string/replace #"%3A" ":")
+      (string/replace #"%0D" "\r")
+      (string/replace #"%0A" "\n")
+      (string/replace #"%25" "%")))
```

Comment:
This is perhaps finicky but I'd love these escape/unescape to be more declarative.

I believe that this will work:

```clojure
(def ^:private anvl-replacements
  {"\n" "%0A"
   "\r" "%0D"
   "%" "%25"
   ":" "%3A"})

(defn- anvl-escape [value]
  (reduce #(string/replace %1 (re-pattern (first %2)) (second %2)) value anvl-replacements))

(defn- anvl-unescape [value]
  (reduce #(string/replace %1 (re-pattern (second %2)) (first %2)) value anvl-replacements))
```

I wasn't completely sure this would work with `\r` and `\n` but repl testing seems to suggest it will: 

```
=> (reduce #(string/replace %1 (re-pattern (second %2)) (first %2)) "a%3Aoe%25ui%0D%0A" {"\n" "%0A" "\r" "%0D" "%" "%25" ":" "%3A"})
"a:oe%ui\r\n"
```